### PR TITLE
[FLINK-24454][table-planner][tests] Consolidate cast tests

### PR DESCRIPTION
**Pull request by matriv**
Check for `CAST` tests in other classes, move them in `CastFunctionITCase`,
or completely remove them if are already contained in `CastFunctionITCase`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no
## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by matriv** [comment]
@twalthr Please check again, I created 2 extra hotfix commits, I'll squash the rest once it's approved.

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionITCase.java [hunk 1]
```diff
@@ -88,9 +130,10 @@ protected Configuration configuration() {
                         // https://issues.apache.org/jira/browse/FLINK-24413 - Trim to precision
                         // in this case down to 3 chars
                         .fromCase(STRING(), "abcdef", "abcdef") // "abc"
-                        .fromCase(DATE(), LocalDate.parse("2021-09-30"), "2021-09-30") // "202"
+                        .fromCase(DATE(), DEFAULT_DATE, "2021-09-24") // "202"
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionITCase.java:133]: What was the `202` referring to?

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionITCase.java [hunk 2]
```diff
@@ -979,6 +1265,30 @@ private TestSpec build() {
                     .testResult(testSpecs.toArray(new ResultSpec[0]));
             return testSpec;
         }
+
+        private String tableApiSpecificErrorMsg(int i, LogicalType colType, String defaultMsg) {
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionITCase.java:1269]: this looks like duplicate code to me that can be merged with the method below? also `i` is not used.

**Review by twalthr — APPROVED** [review]
Thanks for the update @matriv. I had 2 minor comments. Will merge this in the next iteration.

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionITCase.java (hunk 1 above)
Comment (matriv) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionITCase.java:133]: "trim to 3 chars" due to the `CHAR(3)`

**Review by twalthr — APPROVED** [review]
LGTM will merge once build is green

**Pull request closed without merging**